contracts/Tellor.sol

```diff
@@ -81,12 +81,46 @@ contract Tellor is TellorStake {
      * @param _destination is the address that will receive tokens
      * @param _amount iis the amount to mint to the user
      */
-    function migrateContract(
+    function migrateFrom(
         address _origin,
         address _destination,
         uint256 _amount
     ) external {
         require(msg.sender == addresses[_MIGRATOR], "not allowed");
+        _migrateFrom(_origin, _destination, _amount);
+    }
+
+    /**
+     * @dev This is an internal function used by the function migrate  that helps to
+     *  swap old trb tokens for new ones based on the user's old Tellor balance
+     * @param _origin is the address of the user to migrate the balance from
+     * @param _destination is the address that will receive tokens
+     * @param _amount iis the amount to mint to the user
+     */
+    function migrateFromBatch(
+        address[] calldata _origin,
+        address[] calldata _destination,
+        uint256[] calldata _amount
+    ) external {
+        require(msg.sender == addresses[_MIGRATOR], "not allowed");
+        require(_origin.length == _destination.length && _origin.length == _amount.length, "mismatching input");
+        for (uint256 index = 0; index < _origin.length; index++) {
+            _migrateFrom(_origin[index], _destination[index], _amount[index]);
+        }
+    }
+
+    /**
+     * @dev This is an internal function used by the function migrate  that helps to
+     *  swap old trb tokens for new ones based on the user's old Tellor balance
+     * @param _origin is the address of the user to migrate the balance from
+     * @param _destination is the address that will receive tokens
+     * @param _amount iis the amount to mint to the user
+     */
+    function _migrateFrom(
+        address _origin,
+        address _destination,
+        uint256 _amount
+    ) internal {
         require(!migrated[_origin], "already migrated");
         _doMint(_destination, _amount);
         migrated[_origin] = true;
```

Comment:
I think we have to take this out for the dexBlue case...can you add a function to lock it or unlock it once we're done with the migration

Reply:
and with this, if you take this part out, you can just remove the migrateFrom and all can go through the migrateFor

Reply:
I honestly think that this is adding too much logic to tellor. My solution would be to migrate all dexBlue balance to another account, or contract, and then do the distribution to holders from there(it'll be probably cheaper too)

This way we can keep a strict rule that no address migrates twice

Reply:
Perfect.